saleor/graphql/attribute/types.py

```diff
@@ -655,34 +655,33 @@ class Meta:
         doc_category = DOC_CATEGORY_ATTRIBUTES
 
 
-class SelectedAttribute(graphene.Interface):
+class SelectedAttribute(ChannelContextTypeForObjectType):
     attribute = graphene.Field(
         Attribute,
         default_value=None,
         description=AttributeDescriptions.NAME,
         required=True,
     )
     values = NonNullList(
-        AttributeValue,
-        description="Values of an attribute.",
-        required=True,
-        deprecation_reason="Use type-specific fields instead.",
+        AttributeValue, description="Values of an attribute.", required=True
     )
 
     class Meta:
         doc_category = DOC_CATEGORY_ATTRIBUTES
-        description = "Represents an assigned attribute to an object."
+        description = "Represents a custom attribute."
 
-    @staticmethod
-    def resolve_type(instance: SelectedAttributeData, _info):
-        if instance.attribute.node.input_type == AttributeInputType.SINGLE_REFERENCE:
-            entity_type = cast(str, instance.attribute.node.entity_type)
-            return ASSIGNED_SINGLE_REFERENCE_MAP.get(entity_type)
-        if instance.attribute.node.input_type == AttributeInputType.REFERENCE:
-            entity_type = cast(str, instance.attribute.node.entity_type)
-            return ASSIGNED_MULTI_REFERENCE_MAP.get(entity_type)
-        attr_type = ASSIGNED_ATTRIBUTE_MAP[instance.attribute.node.input_type]
-        return attr_type
+
+class AssignedAttributeInterface(graphene.Interface):
+    attribute = graphene.Field(
+        Attribute,
+        default_value=None,
+        description="Attribute assigned to an object.",
+        required=True,
+    )
+
+    class Meta:
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+        description = "Represents a single assigned attribute to an object."
```

Comment:
I think a more correct version would be:
```suggestion
        description = "Represents a single attribute assigned to an object."
```